scipy/signal/_waveforms.py

```diff
@@ -372,11 +395,17 @@ def chirp(t, f0, t1, f1, method='linear', phi=0, vertex_zero=True):
 
     We'll use this function to plot the spectrogram in each example.
 
-    >>> def plot_spectrogram(title, w, fs):
-    ...     ff, tt, Sxx = spectrogram(w, fs=fs, nperseg=256, nfft=576)
+    >>> def plot_spectrogram(title, w, fs, one_sided=True):
+    ...     g_std = 12  # standard deviation for Gaussian window in samples
+    ...     win = gaussian(50, std=g_std, sym=True)
+    ...     fft_mode = 'onesided' if one_sided else 'centered'
+    ...     SFT = ShortTimeFFT(win, hop=2, fs=fs, mfft=800, scale_to='psd',fft_mode=fft_mode)
+    ...     Sx2 = SFT.spectrogram(w)
+    ...     Sx_dB = 10 * np.log10(np.fmax(Sx2, 1e-4))
     ...     fig, ax = plt.subplots()
-    ...     ax.pcolormesh(tt, ff[:145], Sxx[:145], cmap='gray_r',
-    ...                   shading='gouraud')
+    ...     im1 = ax.imshow(Sx_dB, origin='lower', aspect='auto',
+    ...             extent=SFT.extent(len(w)), cmap='plasma')
+    ...     fig.colorbar(im1, label = "Power (dB)")
```

Comment:
Perhaps using linear instead of dB scaling is little more suitable here, since it marginally simplifies the code, i.e.:

```python 
Sx2 = SFT.spectrogram(w)
fig, ax = plt.subplots()
im1 = ax.imshow(Sx2, origin='lower', aspect='auto',
                extent=SFT.extent(len(w)), cmap='plasma')
fig.colorbar(im1, label = "Power Spectral Density")
```
Nitpick: The option `scale_to='psd'` produces a power spectral density not a power as written in the colorbar label.